core/domain/exp_jobs_one_off.py

```diff
@@ -628,3 +637,57 @@ def reduce(key, values):
         # Combine all values from multiple lists into a single list
         # for that error type.
         yield (key, list(set().union(*final_values)))
+
+
+class ImageDataMigrationJob(jobs.BaseMapReduceOneOffJobManager):
+    """One-off job for migrating the images in the exploration
+    from the GAE to GCS.
+    """
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [file_models.FileSnapshotContentModel]
+
+    @staticmethod
+    def map(file_snapshot_content_model):
+        # This job is allowed to run only in Production environment since it
+        # uses GcsFileSystem which can't be used in Development environment.
+        if not feconf.DEV_MODE:
+            instance_id = (
+                file_snapshot_content_model.get_unversioned_instance_id())
+            filetype = instance_id[instance_id.rfind('.') + 1:]
+            # To separate the image entries from the audio entries we get from
+            # the FileSnapshotContentModel.
+            if filetype in ALLOWED_IMAGE_EXTENSIONS:
+                pattern = re.compile(
+                    r'^/([^/]+)/assets/(([^/]+)\.(' + '|'.join(
+                        ALLOWED_IMAGE_EXTENSIONS) + '))$')
+                catched_groups = pattern.match(instance_id)
+                if not catched_groups:
+                    yield(WRONG_INSTANCE_ID, instance_id)
+                else:
+                    filename = catched_groups.group(2)
+                    filepath = 'assets/' + filename
+                    exploration_id = catched_groups.group(1)
+                    file_model = file_models.FileModel.get_model(
+                        exploration_id, filepath, False)
+                    if file_model:
+                        content = file_model.content
+                        fs = fs_domain.AbstractFileSystem(
+                            fs_domain.GcsFileSystem(exploration_id))
+                        if fs.isfile('image/%s' % (filename)):
+                            yield(FILE_ALREADY_EXISTS, file_model.id)
+                        else:
+                            fs.commit(
+                                'ADMIN', 'image/%s' % (filename),
```

Comment:
When there's just a single string interpolation arg, no need for parens. Ditto next line.

/cc @apb7

Reply:
Done!